lib/src/impl.dart

```diff
@@ -0,0 +1,25 @@
+import 'dart:async';
+
+abstract class LocalStorageImpl {
+  LocalStorageImpl(this.fileName, [this.path]);
+
+  final String path, fileName;
+
+  Stream<Map<String, dynamic>> get stream;
+
+  void dispose();
+
+  Future init([Map<String, dynamic> initialData]);
```

Comment:
All these `Future` should have a return type. By default it is `dynamic`, you probably want `void`